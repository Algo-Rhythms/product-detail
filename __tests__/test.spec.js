import App from '../client/src/app.jsx';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Container, Jumbotron } from 'react-bootstrap';

const wrapper = mount(<App />);

describe('App', () => {
  test('simple react component test', () => {
    expect(wrapper.find('h1').text()).toEqual('The Proto Company');
  })
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the inner Containers', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Container).length).toEqual(2);
  });
  it('renders the Jumbotron', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Jumbotron).length).toEqual(1);
  });
})
