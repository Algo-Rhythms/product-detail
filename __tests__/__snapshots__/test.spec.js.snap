// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="jumbotron"
  >
    <h1>
      The Proto Company
    </h1>
    <p>
      This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
    </p>
    <p>
      <button
        className="btn btn-primary"
        disabled={false}
        type="button"
      >
        Learn More
      </button>
    </p>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
        s={2}
      >
         
      </div>
      <div
        className="col-12"
        s={8}
      >
         
      </div>
      <div
        className="col"
        s={2}
      >
         
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
        s={2}
      >
         
      </div>
      <div
        className="col-12"
        s={6}
      >
        <div
          className="carousel slide"
          onKeyDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <ol
            className="carousel-indicators"
          >
            <li
              className="active"
              onClick={[Function]}
            />
            <li
              onClick={[Function]}
            />
            <li
              onClick={[Function]}
            />
            <li
              onClick={[Function]}
            />
            <li
              onClick={[Function]}
            />
          </ol>
          <div
            className="carousel-inner"
          >
            <div
              className="active carousel-item"
            >
              <img
                src="https://www.logistec.com/wp-content/uploads/2017/12/placeholder.png"
              />
            </div>
            <div
              className="carousel-item"
            >
              <img
                src="https://www.logistec.com/wp-content/uploads/2017/12/placeholder.png"
              />
            </div>
            <div
              className="carousel-item"
            >
              <img
                src="https://www.logistec.com/wp-content/uploads/2017/12/placeholder.png"
              />
            </div>
            <div
              className="carousel-item"
            >
              <img
                src="https://www.logistec.com/wp-content/uploads/2017/12/placeholder.png"
              />
            </div>
            <div
              className="carousel-item"
            >
              <img
                src="https://www.logistec.com/wp-content/uploads/2017/12/placeholder.png"
              />
            </div>
          </div>
          <a
            className="carousel-control-prev"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
              className="carousel-control-prev-icon"
            />
            <span
              className="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            className="carousel-control-next"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
              className="carousel-control-next-icon"
            />
            <span
              className="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        className="col-12"
        s={4}
      >
        Product Details Here :D
      </div>
      <div
        className="col"
        s={2}
      >
         
      </div>
    </div>
  </div>
</div>
`;
